### PR TITLE
[@types/chrome] Define chrome.declarativeNetRequest

### PR DESCRIPTION
**Pull request by idan315**
* Define the chrome.declarativeNetRequest API
* Added tests

- [✓] Use a meaningful title for the pull request. Include the name of the package modified.
- [✓] Test the change in your own code. (Compile and run.)
- [✓] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [✓] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [✓] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [✓] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [✓] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest

### CODE REVIEW

File: types/chrome/index.d.ts
Comment (JasonXian) [inline types/chrome/index.d.ts:10254]: I think `filter ?: MatchedRulesFilter` over `| undefined` for optional fields? Otherwise we need to explicitly pass undefined, instead of being able to just pass in one argument.

**Review by JasonXian — APPROVED** [review]
LGTM! Thanks for adding this huge API, please address one small thing before merge.

File: types/chrome/index.d.ts
Comment (JasonXian) [reply types/chrome/index.d.ts:10254]: Ah okay that makes sense, merging as is sounds good.

Comment (idan315) [reply types/chrome/index.d.ts:10254]: done. will wait for the checks to conclude and merge it afterwards.

Comment (idan315) [reply types/chrome/index.d.ts:10254]: whoops, now it needs another owner approval. :-)

**Review by JasonXian — APPROVED** [review]

**Comment by idan315** [comment]
Ready to merge

**Pull request merged**
